### DISCUSSION TITLE
Configuration variable list incomplete

### DISCUSSION THREAD

**Discussion by ovizii** [mathesar-foundation/mathesar#2849] [User Q&A]
This list: https://github.com/centerofci/mathesar/blob/develop/docs/docs/install/configuration.md seems to be lacking any information on DJANGO_DATABASE_KEY which is found inside https://github.com/centerofci/mathesar/raw/master/.env.example

**Comment by ovizii** [comment]
Also, the .env.example file only contains this variable:

`MATHESAR_DATABASES=(mathesar_tables|postgresql://mathesar:mathesar@mathesar_db:5432/mathesar)`

one might think one can just change the password to a more complex one:

`MATHESAR_DATABASES=(mathesar_tables|postgresql://mathesar:newsecretpw@mathesar_db:5432/mathesar)`

but actually you need to add this variable to accomplish it:

`POSTGRES_PASSWORD=newsecretpw`

**Comment by ovizii** [comment]
I might be completely wrong, maybe POSTGRES_PASSWORD refers to DJANGO_DATABASE_URL and not to MATHESAR_DATABASES? Who knows :-(

**Comment by silentninja** [comment]
Thank you for initiating this discussion; it's been very beneficial.

> This list: https://github.com/centerofci/mathesar/blob/develop/docs/docs/install/configuration.md seems to be lacking any information on DJANGO_DATABASE_KEY which is found inside https://github.com/centerofci/mathesar/raw/master/.env.example

I agree `DJANGO_DATABASE_KEY` is not documented. #2848 removes the superfluous environment variables.

**Comment by kgodey** [comment]
@ovizii Thanks for the feedback, this is very helpful! I recommend waiting until we release 0.1.2 to look through the documentation. We're working on individual pieces of documentation at a time right now and we're going to do some testing and improvements to make sure it's all coherent before we do the release.

**Comment by ovizii** [comment]
Thanks for taking my comments the right way. I'm just trying to give you feedback, I understand I'm rushing a bit ahead, but please take your time. Lets leave this discussion open and I'll have another look in a few weeks time.

**Comment by kgodey** [comment]
@ovizii Feedback is always welcome! Thank you :)

**Reply by ovizii** [reply]
Sure, but I'm basically complaining that I am not spoon-fed ;-) 

Btw. is there a particular reason why you are using 
`MATHESAR_DATABASES=(mathesar_tables|postgresql://mathesar:mathesar@mathesar_db:5432/mathesar)`
in the .env.sample 

seeing that you also have POSTGRES_PORT, POSTGRES_DB, POSTGRES_USER and POSTGRES_PASSWORD? 
I would have expected to see the latter 4 variables defined in .env.sample and then docker-compose using those variables to puzzle it together like so:
`MATHESAR_DATABASES=(mathesar_tables|postgresql://${POSTGRES_USER: -mathesar }:${POSTGRES_PASSWORD: -mathesar}@mathesar_db:${POSTGRES_PORT: -5432}/${POSTGRES_DB: -mathesar})`

Same could go for:
`DJANGO_DATABASE_URL=postgres://mathesar:mathesar@mathesar_db:5432/mathesar_django`

As far as I know (and I'm an amateur) this is the way to pull a variable and also present a default value at the same time:
`${POSTGRES_DB: -mathesar}`
